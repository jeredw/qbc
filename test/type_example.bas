TYPE Card
    Suit AS STRING * 9
    Value AS INTEGER
END TYPE
DIM Deck(1 TO 52) AS Card
Deck(1).Suit = "Club"
Deck(1).Value = 2
PRINT Deck(1).Suit, Deck(1).Value
