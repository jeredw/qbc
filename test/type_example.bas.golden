(program statement 
  (type_statement TYPE 
    (untyped_id Card) \n 
    (type_element 
      (untyped_id Suit) AS 
      (type_name_for_type_element 
        (fixed_string STRING * 9)) \n) 
    (type_element 
      (untyped_id Value) AS 
      (type_name_for_type_element INTEGER) \n) END TYPE) \n 
  (statement 
    (dim_statement DIM 
      (dim_variable 
        (untyped_id Deck) 
        (dim_array_bounds 
          ( 
            (dim_subscript 
              (expr 
                (literal 1)) TO 
              (expr 
                (literal 52))) )) AS 
        (type_name 
          (untyped_id Card))))) \n 
  (statement 
    (assignment_statement 
      (variable_or_function_call Deck 
        (args_or_indices 
          ( 
            (expr 
              (literal 1)) )) . Suit) = 
      (expr 
        (literal "Club")))) \n 
  (statement 
    (assignment_statement 
      (variable_or_function_call Deck 
        (args_or_indices 
          ( 
            (expr 
              (literal 1)) )) . Value) = 
      (expr 
        (literal 2)))) \n 
  (statement 
    (print_statement PRINT 
      (print_args 
        (expr 
          (variable_or_function_call Deck 
            (args_or_indices 
              ( 
                (expr 
                  (literal 1)) )) . Suit)) , 
        (print_args 
          (expr 
            (variable_or_function_call Deck 
              (args_or_indices 
                ( 
                  (expr 
                    (literal 1)) )) . Value)))))) \n)
